### ISSUE TITLE
Next.js v12 compatibility

### ISSUE THREAD

**Issue by PabloSzx** [shuding/nextra#233]
@shuding 

Clone https://github.com/shuding/nextra and update nextjs to v12.0.0

```json
{
  "next": "^12.0.0"
}
``` 

```
thread '<unnamed>' panicked at 'not yet implemented: proper error reporting for wrong `@jsxRuntime`', /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/swc_ecma_transforms_react-0.55.0/src/jsx/mod.rs:289:25
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at 'not yet implemented: proper error reporting for wrong `@jsxRuntime`', /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/swc_ecma_transforms_react-0.55.0/src/jsx/mod.rs:289:25
error - ./pages/index.mdx
Error: failed to process not yet implemented: proper error reporting for wrong `@jsxRuntime`
(node:82688) [DEP_WEBPACK_MODULE_ISSUER] DeprecationWarning: Module.issuer: Use new ModuleGraph API
(Use `node --trace-deprecation ...` to show where the warning was created)
info  - automatically enabled Fast Refresh for 1 custom loader
wait  - compiling /_error...
thread '<unnamed>' panicked at 'not yet implemented: proper error reporting for wrong `@jsxRuntime`', /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/swc_ecma_transforms_react-0.55.0/src/jsx/mod.rs:289:25
thread '<unnamed>' panicked at 'not yet implemented: proper error reporting for wrong `@jsxRuntime`', /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/swc_ecma_transforms_react-0.55.0/src/jsx/mod.rs:289:25
error - ./pages/index.mdx
Error: failed to process not yet implemented: proper error reporting for wrong `@jsxRuntime`

```

**Comment by wendellhu95** [comment]
I believe this had been fixed by https://github.com/swc-project/swc/issues/2560.

**Comment by PabloSzx** [comment]
Fixed on next@12.0.2 🎉
